packages/local_auth/local_auth_android/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java

```diff
@@ -310,11 +241,10 @@ private void setServicesFromActivity(Activity activity) {
   }
 
   @Override
-  public void onAttachedToActivity(ActivityPluginBinding binding) {
+  public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
     binding.addActivityResultListener(resultListener);
     setServicesFromActivity(binding.getActivity());
     lifecycle = FlutterLifecycleAdapter.getActivityLifecycle(binding);
-    channel.setMethodCallHandler(this);
```

Comment:
We're no longer creating and removing the channel as part of the activity lifecycle. It's inconvenient to do with Pigeon (since we need the messenger), and I don't see any need for it since we're checking the activity anyway. In fact, I think with the old code people would see "missing plugin" exceptions instead of our "this plugin needs a foreground activity" errors, which seems strictly worse.